### PR TITLE
new react getDerivedStateFromProps

### PR DESCRIPTION
**Pull request by rvdende**
Just a simple quick fix to get rid of the react 17.x messages. If you guys think this is not the best way to resolve it and would rather go the componentDidUpdate route feel free to delete this PR.

### CODE REVIEW

**Comment by STRML** [comment]
Would be good to get this to `static getDerivedStateFromProps(props, prevProps)`.

**Comment by rvdende** [comment]
@STRML cool I've updated the PR to use `static getDerivedStateFromProps(props, state).` I've not managed to get yarn to stop throwing me with errors though. Perhaps its time to move this package over to typescript? It is strange to have types in `.js` files for me?

**Comment by daynin** [comment]
>  It is strange to have types in .js files for me

this package uses flowtype (not typescript), that's why we use .js extension instead of .ts

**Comment by rvdende** [comment]
ah I see, thanks for the comment @daynin

File: lib/Resizable.js [hunk 1]
```diff
@@ -114,15 +114,17 @@ export default class Resizable extends React.Component<Props, State> {
...
     // If parent changes height/width, set that in our state.
-    if (!this.state.resizing &&
-        (nextProps.width !== this.props.width || nextProps.height !== this.props.height)) {
-      this.setState({
-        width: nextProps.width,
-        height: nextProps.height
-      });
-    }
+    if (!state.resizing &&
+        (props.width !== props.width || props.height !== props.height)) {
```
Comment (daynin) [inline lib/Resizable.js]: It won't work

```javascript
props.width !== props.width // false (always, except cases with NaN)
```

you should use componentDidMount here

File: lib/ResizableBox.js [hunk 2]
```diff
@@ -35,13 +35,15 @@ export default class ResizableBox extends React.Component<ResizableProps, State>
...
   };
 
-  componentWillReceiveProps(nextProps: ResizableProps) {
-    if (nextProps.width !== this.props.width || nextProps.height !== this.props.height) {
-      this.setState({
-        width: nextProps.width,
-        height: nextProps.height
-      });
+  static getDerivedStateFromProps(props: ResizableProps, state:Object) {
+    if (props.width !== props.width || props.height !== props.height) {
```
Comment (daynin) [inline lib/ResizableBox.js]: the same case as above

**Comment by PeteJodo** [comment]
The `react-draggable` dependency had a very similar PR and was published as a major version, could this bump that dependency so that it removes all usage of the old react APIs. As of right now, including this change will still trigger warnings in `v16.9` due to the `react-draggable` dependency

**Comment by STRML** [comment]
Closing in favor of #112

**Pull request closed without merging**